SNS/src/Akka.Streams.Sns.Tests/SnsPublishFlowSpec.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using Akka.Streams.Dsl;
+using Akka.Streams.TestKit;
+using Amazon.SimpleNotificationService;
+using Amazon.SimpleNotificationService.Model;
+using FluentAssertions;
+using NSubstitute;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Streams.Sns.Tests
+{
+    public class SnsPublishFlowSpec : Akka.TestKit.Xunit2.TestKit
+    {
+        private readonly ActorMaterializer _materializer;
+
+        public SnsPublishFlowSpec(ITestOutputHelper output)
+            : base(output: output)
+        {
+            _materializer = Sys.Materializer();
+        }
+
+        [Fact]
+        public async Task ItShouldPublishASingleMessageToSns()
+        {
+            var publishRequest = new PublishRequest("topic-arn", "sns-message");
+            var publishResult = new PublishResponse { MessageId = "message-id" };
+
+            var snsService = Substitute.For<IAmazonSimpleNotificationService>();
+            snsService.PublishAsync(Arg.Any<PublishRequest>())
+                .Returns(publishResult);
+
+            var (probe, task) = this.SourceProbe<string>()
+                .Via(SnsPublisher.PlainFlow("topic-arn", snsService))
+                .ToMaterialized(Sink.Seq<PublishResponse>(), Keep.Both)
+                .Run(this._materializer);
+
+            probe.SendNext("sns-message").SendComplete();
+            var actualResult = (await task).FirstOrDefault();
+            actualResult.Should().Be(publishResult);
+            await snsService.Received(1).PublishAsync(Arg.Any<PublishRequest>());
+        }
+
+        [Fact]
+        public void ItShouldPublishMultipleMessagesToSns()
```

Comment:
The first test is using async/await. This could make use of it too.

Reply:
Will fix it later. I want to write a couple of integration tests soon